test/Registry.js

```diff
@@ -1,50 +1,38 @@
 var Registry = artifacts.require("./RegistryApp.sol");
 
 contract('Registry', function (accounts) {
-  describe('#add', function () {
-    it('should add entries', function () {
-      return Registry.deployed().then((instance) => {
-        return instance.add("foo")
-      }).then((receipt) => {
-        assert.isTrue(
-          receipt.logs.filter((log) => log.event === 'EntryAdded').length === 1,
-          'should fire EntryAdded event'
-        )
-      })
-    })
+  let app
+
+  beforeEach(async () => {
+    app = await Registry.new()
+    await app.initialize()
   })
 
-  describe('#remove', function () {
-    it('should remove entries', async function () {
-      const instance = await Registry.deployed()
+  it('should add entries', async function () {
+    const receipt = await app.add("foo")
+    assert.isTrue(
+      receipt.logs.filter((log) => log.event === 'EntryAdded').length === 1,
+      'should fire EntryAdded event'
+    )
+  })
 
-      return instance.add("foo").then((receipt) => {
-        var addedEvent = receipt.logs.filter((log) => log.event === 'EntryAdded')[0]
-        var entryId = addedEvent.args.id
+  it('should remove entries', async function () {
+    const receipt1 = await app.add("foo")
+    var addedEvent = receipt1.logs.filter((log) => log.event === 'EntryAdded')[0]
+    var entryId = addedEvent.args.id
 
-        return instance.remove(entryId)
-      }).then((receipt) => {
-        assert.isTrue(
-          receipt.logs.filter((log) => log.event === 'EntryRemoved').length === 1,
-          'should fire EntryRemoved event'
-        )
-      })
-    })
+    const receipt2 = await app.remove(entryId)
+    assert.isTrue(
+      receipt2.logs.filter((log) => log.event === 'EntryRemoved').length === 1,
+      'should fire EntryRemoved event'
+    )
   })
 
-  describe('#get', function () {
-    it('should get an entry', async function () {
-      const instance = await Registry.deployed()
-
-      return instance.add("foo").then((receipt) => {
-        var addedEvent = receipt.logs.filter((log) => log.event === 'EntryAdded')[0]
-        var entryId = addedEvent.args.id
+  it('should get an entry', async function () {
+    const receipt1 = await app.add("foo")
+    var addedEvent = receipt1.logs.filter((log) => log.event === 'EntryAdded')[0]
+    var entryId = addedEvent.args.id
```

Comment:
We should check that the key is the `keccak256` hash for the data added to the registry.